src/main/java/org/sonarsource/sonarlint/ls/connected/sync/ServerSynchronizer.java

```diff
@@ -0,0 +1,196 @@
+    showOperationResult(failedConnectionIds);
+    triggerAnalysisOfAllOpenFilesInBoundFolders(failedConnectionIds);
+  }
+
+  private void triggerAnalysisOfAllOpenFilesInBoundFolders(Set<String> failedConnectionIds) {
+    bindingManager.forEachBoundFolder((folder, folderSettings) -> {
+      if (!failedConnectionIds.contains(folderSettings.getConnectionId())) {
+        analysisScheduler.analyzeAllOpenFilesInFolder(folder);
+      }
+    });
+  }
+
+  private void showOperationResult(Set<String> failedConnectionIds) {
+    if (failedConnectionIds.isEmpty()) {
+      client.showMessage(new MessageParams(MessageType.Info, "All SonarLint bindings successfully updated"));
+    } else {
+      var connections = String.join(", ", failedConnectionIds);
+      client.showMessage(
+        new MessageParams(MessageType.Error, "Binding update failed for the following connection(s): " + connections + ". Look at the SonarLint output for details."));
+    }
+  }
+
+  private Set<String> tryUpdateConnectionsAndBoundProjectStorages(Map<String, Map<String, Set<String>>> projectKeyByConnectionIdsToUpdate, ProgressFacade progress) {
+    var failedConnectionIds = new LinkedHashSet<String>();
+    projectKeyByConnectionIdsToUpdate.forEach(
+      (connectionId, projectKeys) -> tryUpdateConnectionAndBoundProjectsStorages(projectKeyByConnectionIdsToUpdate, progress, failedConnectionIds, connectionId, projectKeys));
+    return failedConnectionIds;
+  }
+
+  private void tryUpdateConnectionAndBoundProjectsStorages(Map<String, Map<String, Set<String>>> projectKeyByConnectionIdsToUpdate, ProgressFacade progress,
+    Set<String> failedConnectionIds, String connectionId, Map<String, Set<String>> branchNamesByProjectKey) {
+    progress.doInSubProgress(connectionId, 1.0f / projectKeyByConnectionIdsToUpdate.size(), subProgress -> {
+      var endpointParamsAndHttpClient = bindingManager.getServerConfigurationFor(connectionId);
+      if (endpointParamsAndHttpClient == null) {
+        failedConnectionIds.add(connectionId);
+        return;
+      }
+      var engineOpt = bindingManager.getOrCreateConnectedEngine(connectionId);
+      if (engineOpt.isEmpty()) {
+        failedConnectionIds.add(connectionId);
+        return;
+      }
+      subProgress.doInSubProgress("Update projects storages", 0.5f, s -> tryUpdateBoundProjectsStorage(
+        branchNamesByProjectKey.keySet(), endpointParamsAndHttpClient, engineOpt.get(), s));
+      subProgress.doInSubProgress("Sync projects storages", 0.5f, s -> syncOneEngine(
+        connectionId, branchNamesByProjectKey, engineOpt.get(), s));
+    });
+  }
+
+  private static void tryUpdateBoundProjectsStorage(Set<String> projectKeys, ServerConnectionSettings.EndpointParamsAndHttpClient endpointParamsAndHttpClient,
+    ConnectedSonarLintEngine engine,
+    ProgressFacade progress) {
+    projectKeys.forEach(projectKey -> progress.doInSubProgress(projectKey, 1.0f / projectKey.length(), subProgress -> {
```

Comment:
I think here is a typo, should be `1.0f / projectKey**s**.length()`

Reply:
Good catch!